crates/sui-core/src/event_handler.rs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::authority::{AuthorityStore, AuthorityStoreWrapper};
+use crate::streamer::Streamer;
+use chrono::prelude::*;
+use move_bytecode_utils::module_cache::SyncModuleCache;
+use std::convert::TryFrom;
+use std::sync::Arc;
+use sui_types::{
+    error::{SuiError, SuiResult},
+    event::{Event, EventEnvelope},
+    messages::TransactionEffects,
+};
+use tokio::sync::mpsc::{self, Sender};
+use tracing::{debug, error};
+
+const EVENT_DISPATCH_BUFFER_SIZE: usize = 1000;
+
+pub fn get_unixtime_ms() -> u64 {
+    let ts_ms = Utc::now().timestamp_millis();
+    u64::try_from(ts_ms).expect("Travelling in time machine")
```

Comment:
Thanks for the change.  BTW I've never seen i64 -> u64 fail so you could probably just do `ts_ms as u64`

Reply:
it can fail when it's negative numbers (hence "travelling in time machine")